### PR TITLE
Prioritized reason code

### PR DESCRIPTION
**Pull request by eperelli**
**Summary**
In fqm-execution-demo, guidance responses with reason details that are not "Missing" or "Present" are prioritized, meaning that they are the first guidance responses listed.

**New behavior**
When running gaps in care, the guidance responses with reason code "Missing" or "Present" are now at the bottom. This was especially important for Bryn's new bundle because guidance responses with other reason codes were at the bottom.

**Code changes**
All code changes were made in DetectedIssueResources.tsx. All of the code to render the accordion remains the same, I just manipulated the guidanceResponseArray so that any guidance responses with reason codes other than "Missing" or "Present" were at the beginning of the array.

**Testing guidance**
I tested this with EX130 although the changes will not be obvious here since the array is already in the desired order. I would still check this to ensure that it is still working. I also tested it with ColorectalCancerScreeningCQM-bundle.json because the array in that bundle is not in the desired order. In addition, using this new bundle meant that I had to upgrade to fqm-execution version 0.5.0 for it to work.

### CODE REVIEW

**Comment by mgramigna** [comment]
So I'm seeing this behavior for the standard dropdown EXM130 bundles, I could be missing something
image

The new bundle that we used as a test case does work though. Maybe we can rework the for loop somehow

File: src/components/Results/DetectedIssueResources.tsx [hunk 1]
```diff
@@ -1,14 +1,27 @@
...
 
 const DetectedIssueResources: React.FC<Props> = ({ detectedIssue }) => {
-  const guidanceResponseArray = fhirpath.evaluate(detectedIssue, 'contained.GuidanceResponse');
+  const guidanceResponses = fhirpath.evaluate(detectedIssue, 'contained.GuidanceResponse');
+  const guidanceResponseArray = [];
+
+  for (let i = 0; i < guidanceResponses.length; i++) {
+    if (
+      fhirpath.evaluate(guidanceResponses[i], 'reasonCode.coding.code') === Enums.CareGapReasonCode.MISSING ||
+      fhirpath.evaluate(guidanceResponses[i], 'reasonCode.coding.code') === Enums.CareGapReasonCode.PRESENT
```
Comment (mgramigna) [inline src/components/Results/DetectedIssueResources.tsx]: Let's pull this `fhirpath.evaluate` into a variable before the `if` so we don't have to repeat the same call twice

Comment (mgramigna) [reply src/components/Results/DetectedIssueResources.tsx]: Ah I think I have more info on why I'm seeing my error. This fhirpath expression gives you an array of strings with one element. Not a string itself. So it is actually always failing this check.

indexing to access the first element of the resulting array from this fhirpath evaluate should do the trick

Comment (eperelli) [reply src/components/Results/DetectedIssueResources.tsx]: Fixed!

File: src/components/Results/DetectedIssueResources.tsx [hunk 2]
```diff
@@ -1,14 +1,25 @@
...
+  const guidanceResponseArray = [];
+
+  for (let i = 0; i < guidanceResponses.length; i++) {
+    const reasonCode = fhirpath.evaluate(guidanceResponses[i], 'reasonCode.coding.code')[0];
+    if (reasonCode === Enums.CareGapReasonCode.MISSING || reasonCode === Enums.CareGapReasonCode.PRESENT) {
+      guidanceResponseArray.push(guidanceResponses[i]);
+    } else {
+      guidanceResponseArray.unshift(guidanceResponses[i]);
+    }
+  }
```
Comment (mgramigna) [inline src/components/Results/DetectedIssueResources.tsx]: Thoughts on switching to a `forEach` here? It's more javascripty

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach

code will remain the same, you just don't need the `i` index

**Review by sarahmcdougall — APPROVED** [review]
Everything works as it should and looks good! I agree with Matt's comment that we may want to use `forEach` for the for loop so that it looks a little cleaner, but that should be an easy fix.

**Review by mgramigna — APPROVED** [review]

**Pull request merged**